### ISSUE TITLE
Unused imports lint should detect redundant prelude imports

### ISSUE THREAD

**Issue by jonas-schievink** [rust-lang/rust#28296]
This code produces no warning:
```rust
use std::vec::Vec;

fn main() {
    Vec::<u8>::new();
}
```

A warning would be useful for old code that imports items which have since made it into the prelude.

**Comment by steveklabnik** [comment]
Since new lints have a big impact on users of rustc, the policy is that they should go through the RFC process like other user-facing changes. As such, I'm going to give this one a close, but if anyone comes across this ticket and wants this lint, consider adding it to clippy and/or writing up an RFC. Thanks!

**Comment by jonas-schievink** [comment]
This isn't just a bug?

**Comment by jonas-schievink** [comment]
FWIW, I still believe that this is just a regular enhancement of the `unused_imports` lint that doesn't require an RFC. Just like #33091.
